test/exec-capabilityboundingset-invert.service

```diff
@@ -0,0 +1,6 @@
+[Unit]
+Description=Test for CapabilityBoundingSet
+
+[Service]
+ExecStart=/bin/sh -c 'c=$(capsh --print | grep "Bounding set " | cut -f 2 -d "="); echo $c; exit $(test -n $c)'
+CapabilityBoundingSet=~
```

Comment:
On Thu, Oct 29, 2015 at 3:26 PM, Evgeny Vereshchagin <
notifications@github.com> wrote:

> In test/exec-capabilityboundingset-invert.service
> <https://github.com/systemd/systemd/pull/1714#discussion_r43390533>:
>
> > @@ -0,0 +1,6 @@
> > +[Unit]
> > +Description=Test for CapabilityBoundingSet
> > +
> > +[Service]
> > +ExecStart=/bin/sh -c 'c=$(capsh --print | grep "Bounding set " | cut -f 2 -d "="); echo $c; exit $(test -n $c)'
> > +CapabilityBoundingSet=~
>
> This is a no-op test. Test passes without CapabilityBoundingSet=~ too.
> Maybe
>
> CapabilityBoundingSet=
> CapabilityBoundingSet=~
>
> ?
>
Hmm, yes but I think it will be better to set
CapabilityBoundingSet=~CAP_CHOWN and test that CAP_CHOWN is not in the
capabilities.


> Also, I think, we should compare: pcaps ppid == pcaps pid.
>

Why? I don't see the benefit of testing this? As long as the binary
executed by the service has the right capabilities it is ok, right?


> Something like:
>
> ExecStart=/bin/bash -c 'parent_caps=$(getpcaps $PPID |& cut -d= -f2); my_caps=$(getpcaps $$$$ |& cut -d= -f2); [[ "$parent_caps" = "$my_caps" ]]'
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/systemd/systemd/pull/1714/files#r43390533>.
>